src/communication/expr.rs

```diff
@@ -0,0 +1,308 @@
+                                module.module_name.clone(),
+                                Some(class.component.container_name.clone()),
+                                method.method_name.clone(),
+                                self.clone(),
+                            ));
+                            break;
+                        }
+                    }
+                    if class
+                        .component
+                        .container_name
+                        .to_lowercase()
+                        .contains(&client_name)
+                        && endpoint_call.is_some()
+                    {
+                        // Prefer this method over any previously chosen ones because
+                        // we found a closely named Service module and updated the endpoint call.
+                        return map_endpoint_call(endpoint_call);
+                    }
+                }
+            }
+        }
+
+        map_endpoint_call(endpoint_call)
+    }
+}
+
+fn match_ident_or(expr: &Expr) -> Option<&Ident> {
+    match expr {
+        Expr::Ident(ident) => Some(ident),
+        _ => None,
+    }
+}
+
+fn get_rpc_service_name(
+    class: Option<&ClassOrInterfaceComponent>,
+    client_ident: &str,
+) -> Option<String> {
+    // Search through the modules pertaining to the client using the client name
+    let mut client_name = client_ident
+        .replace("client", "")
+        .replace("pool", "")
+        .replace("_", "");
+    if let Some(class) = class {
+        for field in class.field_components.iter() {
+            // Go by type information for finding the service name. If not available
+            // use naming convention.
+            let field_name = if field.r#type != "" {
+                field.r#type.to_lowercase()
+            } else {
+                field.field_name.to_lowercase()
+            };
+
+            if field_name.contains("client") {
+                // Strip excess type information and unneeded naming like "client" and underscores
+                let mut field_name = if field_name.contains("client>") {
+                    field_name.split("client>").next()?
+                } else {
+                    field_name.split("client").next()?
+                };
+                if field_name.contains("<") {
+                    field_name = field_name.split("<").last()?;
+                }
+                client_name = field_name.replace("_", "").replace("service", "").into();
+            }
+        }
+    }
+    Some(client_name)
+}
+
+fn get_rest_service_name(
+    class: Option<&ClassOrInterfaceComponent>,
+    arg: Option<&Expr>,
+) -> Option<String> {
+    let config_name = get_rest_config_name(arg?)?;
+    for field in class?.field_components.iter() {
+        if field.field_name == config_name {
+            for annotation in field.annotations.iter() {
+                if annotation.name == "@Value" {
```

Comment:
Why are we using the `Value` annotation to detect services? It's just a Lombok construct for generating a bunch of boilerplate for any class, not necessarily a service class.

Reply:
The reason I was doing that was because of the `${...}` environment variable syntax to get that value if it was possible

Reply:
Basically it's just to see if the argument passed as an argument to a REST call was a field, and if it's a field if it has extra metadata that can be used inside of it

Reply:
For posterity: this was grabbing Spring's `@Value` annotations, not Lombok's.